### PR TITLE
Moving everything from Activity to AppCompatActivity

### PR DESCRIPTION
**Pull request by topherbuckley**

### CODE REVIEW

**Comment by topherbuckley** [comment]
Looks like the error is unrelated to what I added.

```
ioio.examples.holiday.HolidayIOIOSmokeTest > smokeTestSimplyStart[emulator(AVD) - 9] FAILED 
	java.lang.NullPointerException: Attempt to invoke virtual method 'android.hardware.Camera$Parameters android.hardware.Camera.getParameters()' on a null object reference
	at ioio.examples.holiday.HolidayIOIOActivity.onStart(HolidayIOIOActivity.java:176)
Tests on emulator(AVD) - 9 failed: Instrumentation run failed due to 'Process crashed.'
```

Did this error not come up on any other PR after you introduced the smoke tests? I'm not familiar with espresso testing at all, but do we need to provide anything special for emulating camera hardware? I could simply add a nullpointer check on this, but then I'd think the espresso test would lose the ability to fully test the app. Any suggestions other than a simple if not null around this?

**Comment by hannesa2** [comment]
> Did this error not come up on any other PR after you introduced the smoke tests? 

Nope, it's related to this PR. 

Maybe this is a blueprint for your issue https://github.com/hannesa2/LiveEdgeDetection/blob/master/app/src/androidTest/java/info/hannes/liveedgedetection/demo/SmokeTest.kt

**Comment by topherbuckley** [comment]
>Maybe this is a blueprint for your issue https://github.com/hannesa2/LiveEdgeDetection/blob/master/app/src/androidTest/java/info/hannes/liveedgedetection/demo/SmokeTest.kt

I'm not sure what you mean. You think there is something in this smoke test causing the same issue? 

Anyways, I just enclosed everything in a nullpointer check and all the espresso tests pass. Looks like your espresso tester does not have a working virtual camera object. HolidayIOIO built and ran just fine on both my phones I test with (Nexus 5, and Pixel 3a).

**Comment by hannesa2** [comment]
I didn't investiage, but on the first view it looked for a missing accepted camera permission in Espresso test. That's why I  showed you how to solve it

**Comment by topherbuckley** [comment]
> I didn't investigate, but on the first view it looked for a missing accepted camera permission in Espresso test. That's why I showed you how to solve it

Why did you think it was a permission issue? I only saw a null Camera object, so I assumed the espresso test virtual device did not have a camera instance associated with it.

**Comment by hannesa2** [comment]
Every camera needs a permission, and in test it was missing

**Comment by topherbuckley** [comment]
> Every camera needs a permission, and in test it was missing.
> As I wrote: on the first view ...

I'm still not seeing why you think it is missing. It is [here](https://github.com/topherbuckley/ioio/blob/1bb219be4f6fa133844fba0e149c6cded0f8ec72/applications/HolidayIOIO/src/androidTest/java/ioio/examples/holiday/HolidayIOIOSmokeTest.kt#L24) is it not? This looks very similar to what you have in the "blueprint" you linked as well. 

Can you quote something from the [failed test](https://github.com/ytai/ioio/runs/1127116476?check_suite_focus=true) showing what you are talking about with regards to permissions? As I said, I get no results from a string search on the raw log for "permission". 

Regardless, I'm not sure what this has to do with this PR. If there was something broken about the test, nothing in this PR modifies any test files.

File: applications/IOIOManager/src/main/java/ioio/manager/FileReturner.java [hunk 1]
```diff
@@ -31,6 +31,6 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 public interface FileReturner {
-	String ERROR_MESSAGE_EXTRA  = "ioio.manager.ERROR_MESSAGE";
-	int RESULT_ERROR = AppCompatActivity.RESULT_FIRST_USER;
+    String ERROR_MESSAGE_EXTRA = "ioio.manager.ERROR_MESSAGE";
```
Comment (hannesa2) [inline applications/IOIOManager/src/main/java/ioio/manager/FileReturner.java:34]: This change is just noise, in one comit a wrong format and in the other format back.

Do you know these commands ? 
`git commit --fixup <some SHA1>`
and later to squash all once
`git rebase -i master --autosquash`

It's exactly for such a workflow to have a beautiful git history.

**Comment by hannesa2** [comment]
Sorry, as I wrote: I didn't investigate... but it would be better, if I had done it.

I stopped reading after this line `method 'android.hardware.Camera$Parameters android.hardware.Camera.getParameters()'` and made a misleading hint.

But the tests has done something what's his primary purpose. It found an error !

File: applications/IOIOManager/src/main/java/ioio/manager/FileReturner.java (hunk 1 above)
Comment (topherbuckley) [reply applications/IOIOManager/src/main/java/ioio/manager/FileReturner.java:34]: Sorry, I don't think I did that intentionally. It must have been an auto formatting issue with Android Studio. I've seen those pop up from time to time in my commits after mistakenly including them in a `git add .`. This seems to arise from differences in .idea/codeStyles that I never manually change. I will try to take more care in catching these before they are included in my commits. 

I was unaware of that workflow for squashing. Thanks for the tip!

**Comment by topherbuckley** [comment]
I'm actually not really sure how HolidayIOIO is supposed to make use of the camera. Without looking at the code, I think it was supposed to adjust the yellow onboard LED according to the average exposure on the camera or something like that. Anyways I think it is a fairly niche use of the IOIOboard, so I don't think its worth spending too much time figuring out the camera usage in the espresso tests. If someone has a use-case with this in the future we can address it then I suppose.

**Pull request merged**